community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStep.java

```diff
@@ -66,14 +66,24 @@ public void run()
         return 0;
     }
 
+    /**
+     * Called once and signals the start of this step. Responsible for calling {@link #progress(int)}
+     * at least now and then.
+     */
     protected abstract void process();
 
-    protected void itemProcessed()
+    /**
+     * Called from {@link #process()}, reports progress so that statistics are updated appropriately.
+     * @param amount number of items processed since last call to this method.
+     */
+    protected void progress( long amount )
     {
-        if ( ++batch == batchSize )
+        batch += amount;
+        if ( batch >= batchSize )
         {
-            doneBatches.incrementAndGet();
-            batch = 0;
+            int batches = batch / batchSize;
+            doneBatches.addAndGet( batches );
+            batch %= batchSize;
```

Comment:
You can move the modulo up under the divide. Then they can be done as a single instruction, though it probably won't make a difference.